client/src/app/components/forms/inputs/types/docker-input.component.ts

```diff
@@ -0,0 +1,29 @@
+import {Component, OnInit} from "@angular/core";
+
+@Component({
+    selector: 'docker-input',
+    template: `
+         <input name="dockerPull"
+                type="text"
+                class="form-control"
+                id="dockerImage"
+                [(ngModel)]="dockerPull">
+    `,
+})
+export class DockerInput implements OnInit {
```

Comment:
It doesn't have to be. My idea was that if we have an input component that is generic enough to attach different validation or change events, we could use that and pass it the methods we want to execute on these occasions.